### PR TITLE
INTMDB-249: Lint and fix linting for examples

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Link to any related issue(s): INTMDB-249

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: .github/workflows/examples.yml
Comment (gssbzn) [inline .github/workflows/examples.yml:57]: open to suggestions, rules [here](https://github.com/terraform-linters/tflint/tree/master/docs/rules)

**Comment by gssbzn** [comment]
Major liberty I took as we lint for unused declarations, deleted unused vars but disabled the linter for data source examples as I guess those matter.
Other than that most fixed here are:
- use standard comment style
- use modern index access
- some examples were outdated with invalid properties
- some provider versions were not correctly set

**Comment by themantissa** [comment]
@gssbzn thank you for the work on this.   Hopefully with having the internal docs team's help we'll be able to get to a better place but certainly having more help, such as yourself, will go a long way too.

**Comment by themantissa** [comment]
Note: this will have to wait to be reviewed till after we get 1.0.1 out.   DoU staff please do not spend hours on this yet.

**Review by coderGo93 — APPROVED** [review]
:O lots of updates in examples but very helpful for the future PRs if needed, so LGTM, thank you gustavo

File: .github/workflows/examples.yml
Comment (themantissa) [reply .github/workflows/examples.yml:57]: @thetonymaster any input here?

Comment (thetonymaster) [reply .github/workflows/examples.yml:57]: I'll add `terraform_typed_variables` and `terraform_unused_declarations` too. Just to have cleaner code

**Review by thetonymaster — APPROVED** [review]

File: .github/workflows/examples.yml
Comment (themantissa) [reply .github/workflows/examples.yml:57]: @thetonymaster are you asking for a change in this PR, or does the "I'll add" mean you will submit a PR to add these after this is merged?

**Review by themantissa — APPROVED** [review]
LGTM, thank you for doing this.

**Comment by themantissa** [comment]
@nikhil-mongo just FYI

File: .github/workflows/examples.yml
Comment (thetonymaster) [reply .github/workflows/examples.yml:57]: Wrong verb, "`I'd` add"

Comment (gssbzn) [reply .github/workflows/examples.yml:57]: I already added `terraform_unused_declarations`, I'll try to add `terraform_typed_variables` on a separate PR that may be a big change as we are not typing currently and I'd rather merge this sooner rather than later

**Pull request merged**
